### PR TITLE
Finalizing changelog for v5.1

### PR DESCRIPTION
**Pull request by saimn**
### Description

This pull request is to address ...

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Going to close/reopen to make CI happy.

File: CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,447 @@
+Version 5.1 (2022-05-23)
+========================
+
```
Comment (pllim) [inline CHANGES.rst:3]:
```suggestion
```

File: CHANGES.rst [hunk 2]
```diff
@@ -1,3 +1,447 @@
...
+  A supporting equality method is added, and ``repr`` is enhanced to be able to
+  roundtrip -- ``eval(repr(Parameter()))`` -- if the Parameter's arguments can
+  similarly roundtrip.
+  Parameter's arguments are made keyword-only. [#12479]
+
+- Add methods ``Otot`` and ``Otot0`` to FLRW cosmologies to calculate the total
+  energy density of the Universe. [#12590]
+
+- Add property ``is_flat`` to cosmologies to calculate the curvature of the Universe.
+
```
Comment (pllim) [inline CHANGES.rst:36]:
```suggestion
```

**Comment by saimn** [comment]
Ah, just pushed with a `[ci skip]` since we mostly need RTD here ;)

File: CHANGES.rst [hunk 3]
```diff
@@ -1,3 +1,447 @@
...
+  or the specified user, respectively. [#13129]
+
+astropy.io.ascii
+^^^^^^^^^^^^^^^^
+
+- Simplify the way that the ``converters`` argument of ``io.ascii.read`` is
+  provided. Previously this required wrapping each data type as the tuple returned
+  by the ``io.ascii.convert_numpy()`` function and ensuring that the value is a
+  ``list``. With this update you can write ``converters={'col1': bool}`` to force
+  conversion as a ``bool`` instead of the previous syntax ``converters={'col1':
```
Comment (pllim) [inline CHANGES.rst:75]: Is Sphinx smart enough to allow the double backtick with endline in between?

File: CHANGES.rst [hunk 4]
```diff
@@ -1,3 +1,447 @@
...
+
+astropy.modeling
+^^^^^^^^^^^^^^^^
+
+- Add new fitters based on ``scipy.optimize.least_squares`` method of non-linear
+  least-squares optimization:
+
+      - ``TRFLSQFitter`` using the Trust Region Reflective algorithm.
+      - ``LMLSQFitter`` using the Levenberg-Marquardt algorithm (implemented by ``scipy.optimize.least_squares``).
+      - ``DogBoxLSQFitter`` using the dogleg algorithm. [#12051]
```
Comment (pllim) [inline CHANGES.rst:126]: I don't know where one should put the PR number in a list like this. Do we allow sub-list in change log?

File: CHANGES.rst [hunk 5]
```diff
@@ -1,3 +1,447 @@
...
+astropy.wcs
+^^^^^^^^^^^
+
+- ``astropy.wcs.Celprm`` and ``astropy.wcs.Prjprm`` have been added
+  to allow access to lower level WCSLIB functionality and to allow direct
+  access to the ``cel`` and ``prj`` members of ``Wcsprm``. [#12514]
+
+- Add ``temporal`` properties for convenient access of/selection of/testing for
+  the ``TIME`` axis introduced in WCSLIB version 7.8. [#13094]
+
```
Comment (pllim) [inline CHANGES.rst:233]:
```suggestion
```

File: CHANGES.rst [hunk 6]
```diff
@@ -1,3 +1,447 @@
...
+- The already deprecated ``Planck18_arXiv_v2`` has been removed.
+  Use ``Planck18`` instead [#12354]
+
+- ``default_cosmology.get_cosmology_from_string`` is deprecated and will be
+  removed in two minor versions.
+  Use ``getattr(astropy.cosmology, <str>)`` instead. [#12375]
+
+- In I/O, conversions of Parameters move more relevant information from the
+  Parameter to the Column.
+
```
Comment (pllim) [inline CHANGES.rst:258]:
```suggestion
```

File: CHANGES.rst [hunk 7]
```diff
@@ -1,3 +1,447 @@
...
+- The following are deprecated and will be removed in a future release.
+  Use ``pytest`` warning and exception handling instead:
+
+  * ``astropy.io.ascii.tests.common.raises``
+  * ``astropy.tests.helper.catch_warnings``
+  * ``astropy.tests.helper.ignore_warnings``
+  * ``astropy.tests.helper.raises``
+  * ``astropy.tests.helper.enable_deprecations_as_exceptions``
+  * ``astropy.tests.helper.treat_deprecations_as_exceptions`` [#12633]
```
Comment (pllim) [inline CHANGES.rst:334]: Same comment as sub-list comment above.

File: CHANGES.rst [hunk 8]
```diff
@@ -1,3 +1,447 @@
...
+^^^^^^^^^^^^
+
+- Creating an `~astropy.time.TimeDelta` object with numerical inputs
+  that do not have a unit and without specifying an explicit format,
+  for example ``TimeDelta(5)``,
+  now results in a `~astropy.time.TimeDeltaMissingUnitWarning`.
+  This also affects statements like ``Time("2020-01-01") + 5`` or
+  ``Time("2020-01-05") - Time("2020-01-03") < 5``, which implicitly
+  transform the right-hand side into an `~astropy.time.TimeDelta` instance. [#12888]
+
```
Comment (pllim) [inline CHANGES.rst:349]:
```suggestion
```

File: CHANGES.rst [hunk 9]
```diff
@@ -1,3 +1,447 @@
...
+- Updated bundled WCSLIB version to 7.11. This update together with 7.10
+  includes bug fixes to ``tabini()`` and ``tabcpy()`` as well as several
+  print formatting enhancements. For a full list of
+  changes - see http://www.atnf.csiro.au/people/mcalabre/WCS/CHANGES [#13171]
+
+- Fixed error that occurred in ``WCS.world_to_pixel`` for ``WCS`` objects with a
+  spectral axis and observer location information when passing a ``SpectralCoord``
+  that had missing observer or target information. [#13228]
+
+
```
Comment (pllim) [inline CHANGES.rst:443]:
```suggestion
```

**Review by pllim — APPROVED** [review]
I didn't check if entries are in correct section or the wordings; I assume those were checked when PR went in, and there is no bug in the towncrier conversion.

Just some minor comments on the formatting, otherwise LGTM. Thanks!

File: CHANGES.rst (hunk 4 above)
Comment (saimn) [reply CHANGES.rst:126]: A sub-list seems more readable in this case (even if we should to try to keep changelog entries short, in cases like this I think it's better to take a bit of space to explain the changes). Not sure for the PR number though, I would probably put it on the first line.

Comment (pllim) [reply CHANGES.rst]: First line sounds good to me. Thanks!

File: CHANGES.rst (hunk 3 above)
Comment (saimn) [reply CHANGES.rst:73]: Yes it works: https://astropy--13270.org.readthedocs.build/en/13270/changelog.html

**Comment by pllim** [comment]
Feel free to merge if you are happy with this. I have no more comments. Thanks!

**Pull request merged**